ui-speedspacechart/src/components/utils.ts

```diff
@@ -63,26 +63,39 @@ export const getAdaptiveHeight = (
   layersDisplay: Store['layersDisplay'],
   isIncludingLinearLayers: boolean = true
 ): number => {
-  const { ELECTRICAL_PROFILES_HEIGHT, POWER_RESTRICTIONS_HEIGHT, SPEED_LIMIT_TAGS_HEIGHT } =
-    LAYERS_HEIGHTS;
-
-  const layerHeights = {
-    electricalProfiles: ELECTRICAL_PROFILES_HEIGHT,
-    powerRestrictions: POWER_RESTRICTIONS_HEIGHT,
-    speedLimitTags: SPEED_LIMIT_TAGS_HEIGHT,
-  };
-
   let adjustment = 0;
 
-  Object.keys(layerHeights).forEach((key) => {
-    const layer = key as keyof typeof layerHeights;
+  Object.keys(LinearLayerHeightsByName).forEach((key) => {
+    const layer = key as keyof typeof LinearLayerHeightsByName;
     if (layersDisplay[layer]) {
-      adjustment += isIncludingLinearLayers ? layerHeights[layer] : -layerHeights[layer];
+      adjustment += isIncludingLinearLayers
+        ? LinearLayerHeightsByName[layer]
+        : -LinearLayerHeightsByName[layer];
     }
   });
   return height + adjustment;
 };
 
+/**
+ * Calculates the top position for the linear layers based on the height and the layers displayed.
+ * The speed limit tags linear layer needs to call this function with the isSpeedLimitByTag flag set to true.
+ */
+export const getLinearLayerMarginTop = (
+  height: number,
+  layersDisplay: Store['layersDisplay'],
+  isSpeedLimitByTag: boolean = false
+) => {
+  let adjustment = 0;
+  const { electricalProfiles, powerRestrictions, speedLimitTags } = layersDisplay;
+
+  if (electricalProfiles) adjustment = LinearLayerHeightsByName.electricalProfiles;
+
+  if (powerRestrictions && speedLimitTags && isSpeedLimitByTag)
```

Comment:
I believe the `speedLimitTags` condition is unnecessary here. If this is called with `isSpeedLimitByTag === true`, then `speedLimitTags` is necessarily true as well.

Reply:
Good point, done.